src/Take.Blip.Builder/Strategies/CrmContext.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Models;
+using Take.Blip.Builder.Utils;
+
+namespace Take.Blip.Builder.Strategies
+{
+    public class CrmContext : ICrmContext
+    {
+        private ICrmProcessor _crmProcessor;
+        public async Task ExecuteAsync(IContext context, CrmSettings settings, ActionType actionType, CancellationToken cancellationToken)
+        {
+            if (actionType.Equals(ActionType.GetLead))
+            {
+                await _crmProcessor.GetLead(context, settings, cancellationToken);
+            }
+            if (actionType.Equals(ActionType.CreateLead))
+            {
+                await _crmProcessor.RegisterLead(context, settings, cancellationToken);
```

Comment:
RegisterLeadAsync

Reply:
await _crmProcessor.RegisterLead(context, settings, cancellationToken); should be 
await _crmProcessor.RegisterLeadAsync(context, settings, cancellationToken);